core/src/main/scala/org/apache/spark/storage/FallbackStorage.scala

```diff
@@ -158,7 +158,7 @@ object FallbackStorage extends Logging {
         val name = ShuffleDataBlockId(shuffleId, mapId, NOOP_REDUCE_ID).name
         val dataFile = new Path(fallbackPath, s"$appId/$shuffleId/$name")
         val f = fallbackFileSystem.open(dataFile)
-        val size = nextOffset - 1 - offset
+        val size = nextOffset - offset
         logDebug(s"To byte array $size")
```

Comment:
Given this bug, I am wondering if we want to refactor `IndexShuffleBlockResolver` such that we can reuse it here as well ?

Reply:
Maybe, for Apache Spark 3.2.0, @mridulm ? Currently, this PR is aiming to provide this fix for Apache Spark 3.1.0 RC before next Monday.

Reply:
Sounds good, thanks @dongjoon-hyun !

Reply:
Thank you, @mridulm !